packages/flutter_tools/lib/src/android/native_assets.dart

```diff
@@ -0,0 +1,243 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:native_assets_builder/native_assets_builder.dart'
+    show BuildResult, DryRunResult;
+import 'package:native_assets_cli/native_assets_cli.dart' hide BuildMode;
+import 'package:native_assets_cli/native_assets_cli.dart' as native_assets_cli;
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../build_info.dart';
+import '../globals.dart' as globals;
+import '../native_assets.dart';
+import 'android_sdk.dart';
+
+/// Dry run the native builds.
+///
+/// This does not build native assets, it only simulates what the final paths
+/// of all assets will be so that this can be embedded in the kernel file.
+Future<Uri?> dryRunNativeAssetsAndroid({
+  required NativeAssetsBuildRunner buildRunner,
+  required Uri projectUri,
+  bool flutterTester = false,
+  required FileSystem fileSystem,
+}) async {
+  if (!await nativeBuildRequired(buildRunner)) {
+    return null;
+  }
+
+  final Uri buildUri_ = nativeAssetsBuildUri(projectUri, OS.android);
+  final Iterable<Asset> nativeAssetPaths =
+      await dryRunNativeAssetsAndroidInternal(
+          fileSystem, projectUri, buildRunner);
+  final Uri nativeAssetsUri =
+      await writeNativeAssetsYaml(nativeAssetPaths, buildUri_, fileSystem);
```

Comment:
It looks like this file might have been formatted by `dartfmt`. Please read https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-a-maximum-line-length-of-80-characters about how to break up long lines according to the Flutter repo style guide, and apply it here and throughout the PR.
```suggestion
  final Iterable<Asset> nativeAssetPaths = await dryRunNativeAssetsAndroidInternal(
    fileSystem, projectUri, buildRunner,
  );
  final Uri nativeAssetsUri = await writeNativeAssetsYaml(
    nativeAssetPaths, buildUri_, fileSystem,
  );
```